edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -84,18 +90,15 @@ public void InitiateCacheRefresh()
         {
             Events.ReceivedRequestToRefreshCache();
 
-            TimeSpan durationSinceLastRefresh = DateTime.UtcNow - this.cacheLastRefreshTime;
+            DateTime now = DateTime.UtcNow;
```

Comment:
>DateTime now = DateTime.UtcNow; [](http://example.com/codeflow?start=12&length=31)

looking at this logic again, it's prob more correct to have the timestamp to be snapped after the lock in case we end up waiting a while trying to take the lock.